site/src/Consumer/AbstractNotificationConsumer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Consumer;
+
+use App\Util\Notification;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerInterface;
+use Exception;
+
+/**
+ * Class AbstractNotificationConsumer
+ *
+ * Base class for all the Notification Consumers that we have
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+abstract class AbstractNotificationConsumer implements ConsumerInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var Notification $notification
+     */
+    protected $notification;
+
+    /**
+     * AbstractNotificationConsumer constructor.
+     * @param LoggerInterface $logger
+     * @param Notification    $notification
+     */
+    public function __construct(LoggerInterface $logger, Notification $notification)
```

Comment:
Please remove Notification from the constructor, it's not injected or passed there somehow: it's set in the execute() function.